### PR TITLE
Update to support Xcode 10.2

### PR DESCRIPTION
**Pull request by mpdifran**
This also resolves a few warnings that popped up in the new Xcode.

### CODE REVIEW

File: Tests/SwinjectAutoregistrationTests/TypeParserTests.swift [hunk 1]
```diff
@@ -128,14 +128,6 @@ class TypeParserSpec: QuickSpec {
                 XCTAssertEqual(type, Type.closure(parameters: [], returnType: void, throws: true))
             }
             
-            it("scans implicitly unwrapped optional") {
-                let string = "\((String!).self)"
```
Comment (mpdifran) [inline Tests/SwinjectAutoregistrationTests/TypeParserTests.swift]: Latest Xcode does not allow `(String!).self`. This was producing `"Optional<String>"` anyway, which is the case covered by the below test case.

File: SwinjectAutoregistration.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -874,6 +877,7 @@
 				);
 				INFOPLIST_FILE = "$(SRCROOT)/Sources/Info.plist";
 				PRODUCT_BUNDLE_IDENTIFIER = "com.swinject.SwinjectAutoregistration-iOS";
+				SWIFT_VERSION = 5.0;
```
Comment (freak4pc) [inline SwinjectAutoregistration.xcodeproj/project.pbxproj]: This would force consumers of the app to run against Swift 5, should we actually do that?

Comment (mpdifran) [reply SwinjectAutoregistration.xcodeproj/project.pbxproj]: That's a fair point. We should probably keep it at its current value.

File: Sources/AutoRegistration.swift [hunk 3]
```diff
@@ -452,7 +452,7 @@ public func autoregister<Service, A, B, C, D, E, F, Arg1>(_ service: Service.Typ
  - Important: Might fail if one of the dependencies is unresolvable.
  */
 @discardableResult
-public func autoregister<Service, A, B, C, D, E, F, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D, E, F)) -> Service) -> ServiceEntry<Service> {
+    func autoregister<Service, A, B, C, D, E, F, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D, E, F)) -> Service) -> ServiceEntry<Service> {
```
Comment (freak4pc) [inline Sources/AutoRegistration.swift]:
```suggestion
func autoregister<Service, A, B, C, D, E, F, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D, E, F)) -> Service) -> ServiceEntry<Service> {
```

File: Sources/AutoRegistration.swift [hunk 4]
```diff
@@ -474,7 +474,7 @@ public func autoregister<Service, A, B, C, D, E, F, Arg1, Arg2>(_ service: Servi
  - Important: Might fail if one of the dependencies is unresolvable.
  */
 @discardableResult
-public func autoregister<Service, A, B, C, D, E, F, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D, E, F)) -> Service) -> ServiceEntry<Service> {
+    func autoregister<Service, A, B, C, D, E, F, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D, E, F)) -> Service) -> ServiceEntry<Service> {
```
Comment (freak4pc) [inline Sources/AutoRegistration.swift]:
```suggestion
func autoregister<Service, A, B, C, D, E, F, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D, E, F)) -> Service) -> ServiceEntry<Service> {
```

File: Sources/AutoRegistration.swift [hunk 5]
```diff
@@ -388,7 +388,7 @@ public func autoregister<Service, A, B, C, D, E, Arg1, Arg2>(_ service: Service.
  - Important: Might fail if one of the dependencies is unresolvable.
  */
 @discardableResult
-public func autoregister<Service, A, B, C, D, E, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
+    func autoregister<Service, A, B, C, D, E, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
```
Comment (freak4pc) [inline Sources/AutoRegistration.swift]:
```suggestion
func autoregister<Service, A, B, C, D, E, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
```

File: Sources/AutoRegistration.swift [hunk 6]
```diff
@@ -366,7 +366,7 @@ public func autoregister<Service, A, B, C, D, E, Arg1>(_ service: Service.Type,
  - Important: Might fail if one of the dependencies is unresolvable.
  */
 @discardableResult
-public func autoregister<Service, A, B, C, D, E, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
+    func autoregister<Service, A, B, C, D, E, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
```
Comment (freak4pc) [inline Sources/AutoRegistration.swift]:
```suggestion
func autoregister<Service, A, B, C, D, E, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
```

File: Sources/AutoRegistration.swift [hunk 7]
```diff
@@ -324,7 +324,7 @@ public func autoregister<Service, A, B, C, D, Arg1, Arg2, Arg3>(_ service: Servi
  - Important: Might fail if one of the dependencies is unresolvable.
  */
 @discardableResult
-public func autoregister<Service, A, B, C, D, E>(_ service: Service.Type, name: String? = nil, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
+    func autoregister<Service, A, B, C, D, E>(_ service: Service.Type, name: String? = nil, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
```
Comment (freak4pc) [inline Sources/AutoRegistration.swift]:
```suggestion
func autoregister<Service, A, B, C, D, E>(_ service: Service.Type, name: String? = nil, initializer: @escaping ((A, B, C, D, E)) -> Service) -> ServiceEntry<Service> {
```

File: Sources/AutoRegistration.swift [hunk 8]
```diff
@@ -302,7 +302,7 @@ public func autoregister<Service, A, B, C, D, Arg1, Arg2>(_ service: Service.Typ
  - Important: Might fail if one of the dependencies is unresolvable.
  */
 @discardableResult
-public func autoregister<Service, A, B, C, D, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
+    func autoregister<Service, A, B, C, D, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
```
Comment (freak4pc) [inline Sources/AutoRegistration.swift]:
```suggestion
func autoregister<Service, A, B, C, D, Arg1, Arg2, Arg3>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, _ arg3: Arg3.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
```

File: Sources/AutoRegistration.swift [hunk 9]
```diff
@@ -280,7 +280,7 @@ public func autoregister<Service, A, B, C, D, Arg1>(_ service: Service.Type, nam
  - Important: Might fail if one of the dependencies is unresolvable.
  */
 @discardableResult
-public func autoregister<Service, A, B, C, D, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
+    func autoregister<Service, A, B, C, D, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
```
Comment (freak4pc) [inline Sources/AutoRegistration.swift]:
```suggestion
func autoregister<Service, A, B, C, D, Arg1, Arg2>(_ service: Service.Type, name: String? = nil, arguments  arg1: Arg1.Type, _ arg2: Arg2.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
```

File: Sources/AutoRegistration.swift [hunk 10]
```diff
@@ -259,7 +259,7 @@ public func autoregister<Service, A, B, C, D>(_ service: Service.Type, name: Str
  - Important: Might fail if one of the dependencies is unresolvable.
  */
 @discardableResult
-public func autoregister<Service, A, B, C, D, Arg1>(_ service: Service.Type, name: String? = nil, argument  arg1: Arg1.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
+    func autoregister<Service, A, B, C, D, Arg1>(_ service: Service.Type, name: String? = nil, argument  arg1: Arg1.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
```
Comment (freak4pc) [inline Sources/AutoRegistration.swift]:
```suggestion
func autoregister<Service, A, B, C, D, Arg1>(_ service: Service.Type, name: String? = nil, argument  arg1: Arg1.Type, initializer: @escaping ((A, B, C, D)) -> Service) -> ServiceEntry<Service> {
```

**Comment by mpdifran** [comment]
Thanks for making the changes, I've incorporated them to the PR. I totally missed that when I opened the PR.

**Comment by tkohout** [comment]
@mpdifran Hello! Thanks for all the work! The Autoregistration.swift file is dynamically generated, would you mind looking into [bin/generate](https://github.com/Swinject/SwinjectAutoregistration/blob/master/bin/generate) script and changing it accordingly? There is another pull request https://github.com/Swinject/SwinjectAutoregistration/pull/45/files solving this but it seems it changes different lines. 

I am still in process of updating to the newest XCode which might take some time (bad Internet connection), so I don't know what exactly is the problem.

Thanks!

File: bin/generate [hunk 11]
```diff
@@ -227,7 +227,7 @@ var autoregistrationOutput = [
     "let maxDependencies = \(dependenciesCount)",
     "let maxArguments = \(argumentsCount)",
     "\n",
-    "public extension Container {\n",
+    "extension Container {\n",
```
Comment (mpdifran) [inline bin/generate]: I saw this was removed in #45 , does it make sense to remove it in this PR as well?

**Comment by mpdifran** [comment]
Why is Travis taking so long to build?

**Comment by freak4pc** [comment]
I think the problem isn't the build time, but the queueing time 😂

File: bin/generate (hunk 11 above)
Comment (mpdifran) [reply bin/generate]: Looks like we need to keep this, build is failing.

**Comment by mpdifran** [comment]
Looks like the build is succeeding now. The one job failed because we maxed out GitHub API calls. Can someone just run that job again?

**Comment by mpdifran** [comment]
Anything preventing this from getting merged?

**Comment by enoelboostmi** [comment]
Is there anything missing to merge an release this PR by any chance?

**Comment by tkohout** [comment]
Seems good to me, thanks! I will merge this and release new version. We will look into the swift 5 in the next minor version.

**Pull request merged**